File: data/data_original/at_coder/arc009/C/3839933.py
def c_24(N, K):
    MOD = 1777777777

    class Combination(object):
        # ???????????????????
        def __init__(self, mod=10**15 + 91):
            self.mod = mod

        def _mult_inverse_(self, n):
            return pow(n, self.mod - 2, self.mod)

        def comb(self, n, r):
            if n < 0 or r < 0 or n < r:
                return 0
            numerator, denominator = 1, 1
            for k in range(n, n - r, -1):
                numerator = (numerator * k) % self.mod
            for k in range(r, 0, -1):
                denominator = (denominator * k) % self.mod
            return (numerator * self._mult_inverse_(denominator)) % self.mod

    def montmort(n):
        # 1~n ????????? a ?????? a_i != i ?????????????
        # ??????????????????????
        t1, t2 = 0, 1
        for k in range(3, n + 1):
            t1, t2 = t2, ((k - 1) * (t1 + t2)) % MOD
        return t2

    # N??????????????????????comb(N, K)??
    # K???????????????????????????????????
    ans = (Combination(MOD).comb(N, K) * montmort(K)) % MOD
    return ans

N, K = [int(i) for i in input().split()]
print(c_24(N, K))

File: data/data_transpiled/at_coder/arc009/C/3839933.java
public static int c_24 ( int N , int K ) {
  final int MOD = 1777777777 ;
  class Combination {
    int mod ;
    public int mod ;
    public int mod ;
    public int mult_inverse_ ( int n ) {
      return pow ( n , mod - 2 , mod ) ;
    }
    public int comb ( int n , int r ) {
      if ( n < 0 || r < 0 || n < r ) return 0 ;
      int numerator = 1 , denominator = 1 ;
      for ( int k = n ;
      k > n - r ;
      k -- ) {
        numerator = ( numerator * k ) % mod ;
      }
      for ( int k = r ;
      k > 0 ;
      k -- ) {
        denominator = ( denominator * k ) % mod ;
      }
      return t2 ;
    }
  }
  int ans = 0 ;
  for ( String i : input . split ( ) ) {
    ans = Integer . parseInt ( i ) ;
  }
  System . out . println ( c_24 ( N , K ) ) ;
  return ans ;
}
